alfalfa_web/server/api-v2.js

```diff
@@ -304,6 +353,61 @@ router.get("/version", (req, res) => {
   res.json({ version, ...sha });
 });
 
+router.get("/models", async (req, res) => {
+  api
+    .listModels()
+    .then((data) => res.json({ data }))
+    .catch(() => res.sendStatus(500));
+});
+
+router.post("/models/:id/createRun", (req, res) => {
+  const { id: modelID } = req.params;
+  api
+    .createRunFromModel(modelID)
+    .then((runID) => {
+      res.json({ runID: runID });
+    })
+    .catch((err) => {
+      console.log(err);
+      res.sendStatus(500);
+    });
+});
+
+// Create a post url for file uploads
+// from a browser
+router.post("/models/upload", (req, res) => {
+  const { modelName: modelName } = req.body;
+  const modelID = uuidv1();
+  const modelPath = `uploads/${modelID}/${modelName}`;
+
+  api.createModel(modelID, modelName);
+
+  // Construct a new postPolicy.
+  const params = {
+    Bucket: process.env.S3_BUCKET,
+    Fields: {
+      key: modelPath
+    }
+  };
+
+  api.s3.createPresignedPost(params, function (err, data) {
+    if (err) {
+      throw err;
+    } else {
+      // if you're running locally and using internal Docker networking ( "http://minio:9000")
+      // as your S3_URL, you need to specify an alternate S3_URL_EXTERNAL to POST to, ie "http://localhost:9000"
+      if (process.env.S3_URL_EXTERNAL) {
+        data.url = `${process.env.S3_URL_EXTERNAL}/${process.env.S3_BUCKET}`;
+      } else {
+        data.url = `${process.env.S3_URL}/${process.env.S3_BUCKET}`;
+      }
+      data.modelID = modelID;
+      res.send(JSON.stringify(data));
+      res.end();
```

Comment:
Can we simplify this to to `res.json(data)` or do we explicitly want a text/plain response?